core/models.py

```diff
@@ -194,3 +194,41 @@ def __str__(self):
 
     class Meta:
         verbose_name_plural = "Audit log entries"
+
+
+class ScheduledEvent(BaseEntity):
+    """Scheduled Event Class."""
+    interval_type_choices = (
+        ('min', 'Minutes'),
+        ('sec', 'Seconds'),
+        ('hrs', 'Hours'),
+        ('dys', 'Days'),
+        ('wks', 'Weeks'),
+        ('mnt', 'Months'),
+        ('yrs', 'Years'),
+    )
+    name = models.TextField(null=False, blank=False)
+    description = models.TextField(null=False, blank=True)
+    script_name = models.TextField(null=False, blank=False)
+    enabled = models.BooleanField()
+    start_date = models.DateTimeField(null=False, blank=False)
+    repeat = models.BooleanField()
+    interval = models.BigIntegerField(blank=True, null=True)
+    interval_type = models.CharField(max_length=1, choices=interval_type_choices, null=True, blank=True)
```

Comment:
hmm... confused ... max_length=1 for a CharField? And this is working with your choices? Cos all your choices have at least 3 characters...

Reply:
It should be at most 3, now updated